cli/command/trust/key_generate.go

```diff
@@ -0,0 +1,87 @@
+package trust
+
+import (
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/trust"
+	"github.com/docker/notary"
+	"github.com/docker/notary/trustmanager"
+	"github.com/docker/notary/tuf/data"
+	tufutils "github.com/docker/notary/tuf/utils"
+	"github.com/spf13/cobra"
+)
+
+func newKeyGenerateCommand(dockerCli command.Streams) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "key-generate NAME [NAME...]",
+		Short: "Generate a signing key",
+		Args:  cli.RequiresMinArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return generateKeys(dockerCli, args)
+		},
+	}
+	return cmd
+}
+
+func generateKeys(streams command.Streams, keyNames []string) error {
```

Comment:
Any way we could improve the coverage of this function?

Reply:
ok updated - I've made the passphrase retriever mockable so we can test success cases